pkg/scheduler/serviceresource/helper.go

```diff
@@ -22,4 +60,82 @@ func listCandidateConnectors(ctx context.Context, modelClient model.ClientSet) (
 			connector.CategoryNEQ(types.ConnectorCategoryCustom),
 			connector.CategoryNEQ(types.ConnectorCategoryVersionControl)).
 		All(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot list connectors: %w", err)
+	}
+
+	if len(cs) == 0 {
+		return nil, nil, nil
+	}
+
+	// Index the operators by the related connector ID.
+	indexer := make(map[object.ID]optypes.Operator, len(cs))
+
+	// Record the operators' parallelism by the corresponding trait.
+	limiter := make(operatorLimiter)
+
+	for i := range cs {
+		var op optypes.Operator
+
+		op, err = operator.Get(ctx, optypes.CreateOptions{
+			Connector: *cs[i],
+		})
+		if err != nil {
+			// Warn out without breaking the whole syncing.
+			logger.Warnf("cannot get operator of connector %q: %v", cs[i].ID, err)
+			continue
+		}
+
+		if err = op.IsConnected(ctx); err != nil {
+			// Warn out without breaking the whole syncing.
+			logger.Warnf("unreachable connector %q", cs[i].ID)
+			// Replace disconnected connector with unknown connector.
+			op = operator.UnReachable()
+		}
+
+		// Index operators by connector ID.
+		indexer[cs[i].ID] = op
+
+		// Group connector ID list by operator trait.
+		if t, b := op.Trait(), op.Burst(); t != "" && b > 0 {
+			if _, ok := limiter[t]; !ok {
+				limiter[t] = make(chan struct{}, b)
+			}
+		}
+	}
+
+	return indexer, limiter, nil
+}
+
+// getBucket returns the bucket size and count with the given total and burst,
+// the multiple of the returning results must be greater than or equal to the given total,
+// especially, the returning count must be less than or equal to the given burst.
+func getBucket(total, burst, minSize int) (size, count int) {
+	if minSize <= 0 {
```

Comment:
The count parameter is confusing. Because the burst parameter generally means the maximum number of tokens allowed in a bucket, but here brust is applied to the count parameter.

Reply:
updated, modify to `getBatches` and enrich the comment message.